### PR TITLE
feat: Initial infrastructure for MiniWdl support

### PR DESCRIPTION
**Pull request by nbraid**
Issue #, if available: No GH issue

**Description of Changes**

Adds a MiniWdl stack which creates the appropriate batch resources and job definition to run MiniWdl jobs.

**Description of how you validated changes**

Running a local WES adapter, able to call run hello.wdl workflows.
Able to run using `agc workflow run hello -c miniWdlContext`

**Checklist**

- [X] If this change would make any existing documentation invalid, I have included those updates within this PR
- [X] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have linted my code before raising the PR
- [X] Title of this Pull Request follows Conventional Commits standard: https://www.conventionalcommits.org/en/v1.0.0/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+export interface MiniWdlEngineProps {
+  readonly vpc: IVpc;
+  readonly outputBucketName: string;
+  readonly engineBatch: Batch;
+  readonly workerBatch: Batch;
+}
+
+const MINIWDL_IMAGE_DESIGNATION = "miniwdl";
+
+export class MiniWdlEngine extends Construct {
```
[the remark this answers is not in the document]
Comment (nbraid) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts:20]: Sure, I'll extract what I can.

File: packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        jobRole: engineBatch.role,
+        executionRole: engineBatch.role,
+        image: createEcrImage(this, MINIWDL_IMAGE_DESIGNATION),
+        platformVersion: FargatePlatformVersion.VERSION1_4,
+        command: [],
+        environment: {
+          MINIWDL__AWS__FS: fileSystem.fileSystemId,
+          MINIWDL__AWS__FSAP: accessPoint.accessPointId,
+          MINIWDL__AWS__TASK_QUEUE: workerBatch.jobQueue.jobQueueArn,
+          MINIWDL_S3_OUTPUT_URI: `s3://${outputBucketName}/miniwdl`,
```
[the remark this answers is not in the document]
Comment (nbraid) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts:59]: Yeah, not sure if we want this by default or not. Seems like we'd ideally want to introduce something to clean up EFS asynchronously so we can keep the cache around for a while before cycling out old stuff.

Call caching seems like an important feature. Perhaps we should leave this off by default and introduce another mechanism to clean up the EFS contents?

File: packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    this.logGroup = new LogGroup(this, "EngineLogGroup");
+
+    const fileSystem = new FileSystem(this, "FileSystem", {
+      vpc: vpc,
+      encrypted: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    const accessPoint = new AccessPoint(this, "AccessPoint", {
+      fileSystem: fileSystem,
```
[the remark this answers is not in the document]
Comment (nbraid) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts:38]: Is this user set up already by the launch template, or does it need to be added? What would the corresponding `gid` be?

[the remark this answers is not in the document]
Comment (nbraid) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts]: Seems to work with `uid=0, gid=0`

File: packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mlin) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts:45]: You're right -- I don't think we have to solve this right now, but glad we discussed it as I think it'll remain an unresolved tension for the more S3-focused users.

File: packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mlin) [reply packages/cdk/lib/constructs/engines/miniwdl/miniwdl-engine.ts]: Yea setting uid=0, gid=0 for the access point basically just says "ignore all posix permissions" (whilst the file system is secured by IAM permissions and security groups)

**Review by elliot-smith — DISMISSED** [review]

**Review by tneely — APPROVED** [review]

**Review by elliot-smith — DISMISSED** [review]

**Review by tneely — APPROVED** [review]

**Review by elliot-smith — APPROVED** [review]

**Pull request merged**